src/sage/matroids/oriented_matroids/signed_subset_element.py

```diff
@@ -0,0 +1,850 @@
+
+    def negatives(self):
+        """
+        Return the set of negatives.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: M = OrientedMatroid([[1,-1,1],[-1,1,-1]], key='circuit')
+            sage: E = M.elements()[0]
+            sage: E.negatives()
+            {1}
+        """
+        return self._n
+
+    def zeros(self):
+        r"""
+        Return the set of zeros.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: M = OrientedMatroid([[1,-1,0],[-1,1,0]], key='circuit')
+            sage: E = M.elements()[0]
+            sage: E.zeros()
+            {2}
+        """
+        return self._z
+
+    def support(self):
+        r"""
+        Return the support set.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: M = OrientedMatroid([[1,-1,0],[-1,1,0]], key='circuit')
+            sage: E = M.elements()[0]
+            sage: E.support()
+            {0, 1}
+        """
+        return self._p.union(self._n)
+
+    def groundset(self):
+        r"""
+        Return the groundset.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: M = OrientedMatroid([[1,-1,0],[-1,1,0]], key='circuit')
+            sage: E = M.elements()[0]
+            sage: E.groundset()
+            [0, 1, 2]
+        """
+        return self._g
+
+    def composition(self, other):
+        r"""
+        Return the composition of two elements.
+
+        The composition of two elements `X` and `Y`,
+        denoted `X \circ Y` is given componentwise
+        where for `e \in E` we have `(X \circ Y)(e) = X(e)`
+        if `X(e) \neq 0` else it equals `Y(e)`.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: OrientedMatroid.options.display='vector'
+            sage: M = OrientedMatroid([[0],[1],[-1]], key='vector')
+            sage: E1 = M.elements()[0]; E2 = M.elements()[1]
+            sage: E1.composition(E2)
+            (1)
+            sage: E2.composition(E1)
+            (1)
+            sage: E1.composition(E2) == E2.composition(E1)
+            True
+        """
+        p = []
+        n = []
+        z = []
+        for e in self.groundset():
+            x = self(e)
+            # If x is non-zero, keep its value
+            if x == 1:
+                p.append(e)
+            elif x == -1:
+                n.append(e)
+            else:
+                # else grab the value of the other
+                x = other(e)
+                if x == 1:
+                    p.append(e)
+                elif x == -1:
+                    n.append(e)
+                else:
+                    z.append(e)
```

Comment:
```suggestion
        for e in self._g:
            # If x is non-zero, keep its value
            if e in self._p:
                p.append(e)
            elif e in self._n:
                n.append(e)
            else:
                # else grab the value of the other
                if e in other._p:
                    p.append(e)
                elif e in other._n:
                    n.append(e)
                else:
                    z.append(e)
```
Yes, it unrolls things a bit, but I think it is better.